src/box/lua/misc.cc

```diff
@@ -502,7 +466,7 @@ box_lua_misc_init(struct lua_State *L)
 	static const struct luaL_Reg boxlib_internal[] = {
 		{"prepare_auth", lbox_prepare_auth},
 		{"select", lbox_select},
-		{"new_tuple_format", lbox_tuple_format_new},
+		{"new_external_tuple_format", lbox_new_external_tuple_format},
```

Comment:
Is it changed, because this is not a general purpose format creating function?

I would name it based on what does it do rather than according to how it is supposed to be used. Say, `new_typeless_tuple_format`.

Reply:
This function is used only for tuple formats received over IPROTO, hence the name. Maybe we can go for `new_iproto_tuple_format`, but originally we argreed on `external` in peer discussion.

Reply:
Just in case my angle of view was missed in your peer discussion, I'll describe it a bit more and leave you to decide.

I'm looking on this function as on the tuple format constructor. It is okay to have several different constructors for an object. I often see `foo_new()` and `foo_new_from_bar()`, for example.

I think that there is nothing specific to iproto and it is just how the function is used. If it is correct, I think that we can highlight a difference from the primary constructor in the name: 'typeless', 'dict only', 'from names', etc.

If I'm wrong and there is iproto specifics inside, it's ok to use something like 'new_tuple_format_from_iproto', 'new_tuple_format_from_net_box' or, well, 'external'.

BTW, it would be nice to highlight in a comment what exactly is iproto/net.box specific we have there.

BTW (2), I think that a constructor should be placed together with other module functions. There is the separate discussion thread on it and I'll update it as well.